app/models/event_stream.rb

```diff
@@ -82,19 +89,30 @@ def self.group_name(group)
     group.nil? ? 'Other' : group[:name]
   end
 
+  # TODO: Consider moving since this is EmsEvent specific. group, group_level and group_name exposed as a virtual columns for reports/api.
   def group
     return @group unless @group.nil?
     @group, @group_level = self.class.group_and_level(event_type)
     @group
   end
 
+  # TODO: Consider moving since this is EmsEvent specific. group, group_level and group_name exposed as a virtual columns for reports/api.
   def group_level
     return @group_level unless @group_level.nil?
     @group, @group_level = self.class.group_and_level(event_type)
     @group_level
   end
 
+  # TODO: Consider moving since this is EmsEvent specific. group, group_level and group_name exposed as a virtual columns for reports/api.
   def group_name
     @group_name ||= self.class.group_name(group)
   end
+
+  def self.timeline_classes
+    @timeline_classes ||= EventStream.subclasses.select { |e| e.respond_to?(:group_names_and_levels) }
```

Comment:
Could caching this cause issues with code reload since we are caching classes?

Reply:
It's a relatively cheap operation that really should only create 1 new Array object, so I think we don't need caching here.

Reply:
Sure, it was premature.  The caller shouldn't be hitting this constantly and it's not not problem once classes are loaded.